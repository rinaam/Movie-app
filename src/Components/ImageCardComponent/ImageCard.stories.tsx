import React from "react";
import { Story, Meta } from "@storybook/react";

import ImageCardComponent, { IImageCardProps } from "./ImageCardComponent";

export default {
  title: "ImageCard",
  component: ImageCardComponent,
} as Meta;

const Template: Story<IImageCardProps> = (args) => (
  <ImageCardComponent {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  imageUrl: "https://image.tmdb.org/t/p/w780/hKHZhUbIyUAjcSrqJThFGYIR6kI.jpg",
  alt: "alt",
  description:
    "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quod harum dolor fuga nihil voluptate corrupti deleniti doloremque ipsa. Placeat nemo laudantium tempora sint aliquam voluptate assumenda delectus voluptas officia dolorum!",
};
