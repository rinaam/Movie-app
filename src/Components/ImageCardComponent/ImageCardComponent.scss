@use "sass:map";
@import "../../Styles/breakpoints.scss";
@import "../../Styles/variables.scss";
.image-card {
  position: relative;
  box-shadow: map.get($box-shadow, "elevation");
  border-radius: 4px;
  width: 100%;
  height: 400px;
  &-img {
    width: 100%;
    height: 500px;
    background: {
      repeat: no-repeat;
      size: cover;
      position: top center;
    }
  }
  &-description {
    position: absolute;
    left: 0;
    bottom: 0;
    z-index: 2;
    background-color: $white;
    width: 100%;
    padding: map.get($spacing, "s") map.get($spacing, "s") 0px
      map.get($spacing, "s");
    line-height: 1.5;
    box-sizing: border-box;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
    @include breakpoint(smallDevices) {
      width: 50%;
    }
  }
}
