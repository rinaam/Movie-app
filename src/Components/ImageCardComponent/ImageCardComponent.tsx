import React from "react";
import "./ImageCardComponent.scss";
import ImageComponet from "../ImageComponent/ImageComponent";

export interface IImageCardProps {
  imageUrl: string;
  description: string;
  alt: string;
}

const ImageCardComponent: React.FC<IImageCardProps> = ({
  imageUrl,
  description,
  alt,
}) => (
  <div className="image-card">
    <ImageComponet
      imgUrl={imageUrl}
      alt={alt}
      className="image-card-img"
      useBackground
    />

    <p className="image-card-description">{description}</p>
  </div>
);
export default ImageCardComponent;
