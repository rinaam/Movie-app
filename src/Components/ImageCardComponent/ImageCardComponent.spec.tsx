import { create } from "react-test-renderer";
import ImageCardComponent from "./ImageCardComponent";
import React from "react";

describe("ImageCardComponent", () => {
  it("should render ImageCardComponent", () => {
    const wrapper = create(
      <ImageCardComponent description="test" imageUrl="test" alt="test" />
    ).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
