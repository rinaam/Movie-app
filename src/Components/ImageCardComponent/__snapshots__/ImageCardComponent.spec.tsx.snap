// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCardComponent should render ImageCardComponent 1`] = `
<div
  className="image-card"
>
  <div
    className="image-card-img"
    style={
      Object {
        "backgroundImage": "url(\\"test\\")",
      }
    }
  />
  <p
    className="image-card-description"
  >
    test
  </p>
</div>
`;
